Cargo.toml

```diff
@@ -36,26 +36,26 @@ default = [
     "yamux",
 ]
 autonat = ["libp2p-autonat"]
-dcutr = ["libp2p-dcutr", "libp2p-metrics/dcutr"]
+dcutr = ["libp2p-dcutr", "libp2p-metrics?/dcutr"]
 deflate = ["libp2p-deflate"]
 dns-async-std = ["libp2p-dns", "libp2p-dns/async-std"]
 dns-tokio = ["libp2p-dns", "libp2p-dns/tokio"]
 floodsub = ["libp2p-floodsub"]
-identify = ["libp2p-identify", "libp2p-metrics/identify"]
-kad = ["libp2p-kad", "libp2p-metrics/kad"]
-gossipsub = ["libp2p-gossipsub", "libp2p-metrics/gossipsub"]
+identify = ["dep:libp2p-identify", "libp2p-metrics?/identify"]
```

Comment:
If I am not mistaken that `dep:` syntax is implicitly defined whenever a dependency is marked as optional as explained [here](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies). Perhaps I am misinterpreting what you mean.

Reply:
I am referring to this part of the docs:

> If you specify the optional dependency with the dep: prefix anywhere in the [features] table, that disables the implicit feature.

But reading it again now, I think I misunderstood how it works! I thought one had to use `dep:` everywhere to disable the implicit features!

Reply:
Can we verify that all the implicit features created by optional dependencies are gone?

You should be able to do that by making a new crate and try depending on this version of libp2p (probably best done with a `path` dependency) and see if features like `libp2p-ping` work.

Reply:
Good idea, I ran this test and yes I can confirm that the optional dependencies are no longer pulled in.

Reply:
Just to make sure I understand this correctly, given that we use `dep:` in the line above, implicit features of **all** optional dependencies are removed?

I am still a bit confused why we do it for `identify`, `ping` and `relay`, but not for `dcutr`, `kad` and `gossipsub`.

Reply:
Mentioning @maschad and @thomaseizinger to make sure this doesn't get lost. Accidentally posted on a resolved conversation.

Reply:
Good catch on `dcutr` :) but for `kad`  and `gossipsub` which are only enabled by the `serde` feature if we use `dep:` syntax then there would be no implicit features with the names `libp2p-kad` and ``libp2p-gossipsub` and so it becomes redundant (since they are already optional dependencies)

Reply:
> Good catch on `dcutr` :) but for `kad` and `gossipsub` which are only enabled by the `serde` feature if we use `dep:` syntax then there would be no implicit features with the names `libp2p-kad` and ``libp2p-gossipsub` and so it becomes redundant (since they are already optional dependencies)
> 
> The docs are a bit confusing though, especially for people like myself who are newer to Rust.

I don't quite understand. Shouldn't we also set `libp2p-gossipsub/serde` as a weak dependency to avoid `libp2p-gossipsub` being pulled if only the `serde` feature is activated? Same for `libp2p-kad`.

Perhaps it would be better to better to go back to the original proposal of using the `dep:` syntax everywhere for simplicity reasons.

Reply:
I agree, that may make more sense in terms of a mental model of the dependency tree. I have updated the PR.